### PR TITLE
Create token-plugins-guide.md

### PR DESCRIPTION
**Pull request by tradersnow222**
A comprehensive guide for building with token plugins.

### CODE REVIEW

File: token-plugins-guide.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+   - [ERC20Plugins.sol (Abstract Plugin-enabled ERC20 contract)](#erc20pluginssol-plugin-enabled-erc20-token-contract)
+   - [Plugin.sol (Abstract Plugin contract)](#pluginsol-abstract-plugin-contract)
+
+[Deployed Examples](#deployed-examples)
+
+[Helpful Links](#helpful-links)
+
+## Overview
+
+Token plugins are smart contracts that extend the capabilities of existing ERC20 tokens. They allow for additional accounting to be added to an existing token without the need to deposit your tokens into a contract. **_This is a revolutionary step in the evolution of DeFi_**, as the entire ecosystem has historically relied upon the transfer of tokens in and out of external contracts to achieve enhanced capabilities!
```
Comment (k06a) [inline token-plugins-guide.md:19]: Those plugins can’t extend existing tokens, but I think we can say they extend plugin-enabled tokens or token-enabled token wrappers (ERC20Wrapper extension).

Comment (tradersnow222) [reply token-plugins-guide.md]: Fixed

File: token-plugins-guide.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+[Helpful Links](#helpful-links)
+
+## Overview
+
+Token plugins are smart contracts that extend the capabilities of plugin-enabled ERC20 tokens and plugin-enabled ERC20 wrappers. They allow for additional accounting to be added, without the need to deposit your tokens into a contract. **_This is a revolutionary step in the evolution of DeFi_**, as the entire ecosystem has historically relied upon the transfer of tokens in and out of external contracts to achieve enhanced capabilities! 
+
+By design, token plugins prevent several common attack vectors and inherently create a fundamental level of security within every token incentive mechanism. Users no longer have to trust their funds in an external contract, whether it be for farming, borrowing/lending, or delegating, etc., therefore adding a fundamental layer of security.
+
+## Primary Benefits
+- 100% permissionless, open to all participants.
```
Comment (k06a) [inline token-plugins-guide.md:24]: Need to explain that creation and deployment of plugins can be done permissionless. Then holders of plugin-enabled token can join/add any plugins they want.

File: token-plugins-guide.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+- Risk-free participation: Token plugins can only update with an account’s balance change. They cannot approve, mint, burn, transfer, or overwrite storage.
```
Comment (k06a) [inline token-plugins-guide.md:25]: Rephrase mentioning 2 main DeFi risks avoidance: plugins doesn’t require token depositing neither token approves.

File: token-plugins-guide.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+- Users can connect multiple plugins, allowing for simultaneous participation in multiple incentive programs or governance systems, etc.
+- Implementation is less than 100 lines of code; very simple to adopt.
```
Comment (k06a) [inline token-plugins-guide.md:27]: Not sure, maybe 150? Need to check with `sloc` software.

**Comment by tradersnow222** [comment]
All requested changes have been made

File: token-plugins-guide.md (hunk 4 above)
Comment (tradersnow222) [reply token-plugins-guide.md]: Nice estimate :)

sloc showing:
- ERC20Plugins.sol: 133 lines
- Plugins.sol: 17 lines

File: token-plugins-guide.md [hunk 5]
```diff
@@ -16,15 +16,15 @@
 
 ## Overview
 
-Token plugins are smart contracts that extend the capabilities of plugin-enabled ERC20 tokens and plugin-enabled ERC20 wrappers. They allow for additional accounting to be added, without the need to deposit your tokens into a contract. **_This is a revolutionary step in the evolution of DeFi_**, as the entire ecosystem has historically relied upon the transfer of tokens in and out of external contracts to achieve enhanced capabilities! 
+Token plugins are smart contracts that extend the capabilities of plugin-enabled ERC20 tokens and plugin-enabled ERC20 wrappers. They allow for additional accounting to be added, without the need to deposit your tokens into a contract. **_This is a revolutionary step in the evolution of DeFi_**, as the entire ecosystem has historically relied upon the transfer of tokens in and out of external contracts to achieve enhanced capabilities! The process of creating and deploying plugins is completely open and permissionless, and holders of plugin-compatible ERC20 tokens can freely participate with any plugin of their choosing.
```
Comment (k06a) [inline token-plugins-guide.md:19]: Add “without having their assets under risk” at the end

Comment (tradersnow222) [reply token-plugins-guide.md]: Done

**Pull request closed without merging**